Compilation failure with paths containing ".c"

On Arch Linux, the compilation fails when using yay:

```
make[2]: *** No rule to make target '/home/user/_private.hache/yay/box86/src/box86-0.2.0/src/wrapped/wrappedlibc_private.h', needed by '../src/wrapped/generated/functions_list.txt'.  Stop.
```

It seems the original path `/home/user/.cache/yay` gets converted to `/home/user/_private.hache/yay` by `src/box86-0.2.0/CMakeLists.txt:370:    string(REPLACE ".c" "_private.h" B ${A})` Maybe using regex that only matches ".c" at the end of the string is a solution?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -369,9 +369,12 @@ set(WRAPPEDS
     "${BOX86_ROOT}/src/wrapped/wrappedcrashhandler.c"
 )
 
+# If BOX86_ROOT contains a ".c", the build breaks...
+string(REPLACE ".c" "_private.h" MODROOT ${BOX86_ROOT})
 set(WRAPPEDS_HEAD "${BOX86_ROOT}/src/wrapped/wrappedd3dadapter9_gen.h")
 foreach(A ${WRAPPEDS})
-    string(REPLACE ".c" "_private.h" B ${A})
+    string(REPLACE ".c" "_private.h" C ${A})
+    string(REPLACE "${MODROOT}" "${BOX86_ROOT}" B ${C})
     set(WRAPPEDS_HEAD ${WRAPPEDS_HEAD} ${B})
     set_source_files_properties(${A} PROPERTIES OBJECT_DEPENDS ${B})
 endforeach()

